examples/readEvents/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/starknet.go/rpc"
+	"github.com/NethermindEth/starknet.go/utils"
+
+	setup "github.com/NethermindEth/starknet.go/examples/internal"
+)
+
+const (
+	CONTRACT_ADDRESS = "0x059a943ca214c10234b9a3b61c558ac20c005127d183b86a99a8f3c60a08b4ff" 
+)
+
+var (
+	FROM_BLOCK uint64 = 657100
+	TO_BLOCK   uint64 = 657101
+)
+
+// main entry point of the program.
+//
+// It initializes the environment and establishes a connection with the client.
+// It then retrieves events from the contract and prints how many it found with
+// a series of more selective filters (all event types, just 2 event types, just
+// those 2 event types but with a specified key value).
+
+// Parameters:
+//
+//	none
+//
+// Returns:
+//
+//	none
+func main() {
+	fmt.Println("Starting readEvents example")
+
+	// Load variables from '.env' file
+	rpcProviderUrl := setup.GetRpcProviderUrl()
+
+	// Initialize connection to RPC provider
+	provider, err := rpc.NewProvider(rpcProviderUrl)
+	if err != nil {
+		panic(fmt.Sprintf("Error dialing the RPC provider: %s", err))
+	}
+
+	fmt.Println("Established connection with the RPC provider")
+
+	// contractAddress is the address of the contract whose events we want to read
+	contractAddress, err := utils.HexToFelt(CONTRACT_ADDRESS)
+	if err != nil {
+		msg := fmt.Errorf("failed to create felt from the contract address %s, error %w", CONTRACT_ADDRESS, err)
+		panic(msg)
+	}
+
+	// fromBlock and toBlock are the block numbers between which we want to read events
+	fromBlock := &FROM_BLOCK
+	toBlock := &TO_BLOCK
+
+	// create an EventFilter that specifies which events we want
+	eventFilter := rpc.EventFilter{
+		FromBlock: rpc.BlockID{Number: fromBlock},
```

Comment:
We can change it to use the `rpc.WithBlockNumber` function, that way we don't need the FROM/TO_BLOCK variables